ApplyToBecomeInternal/ApplyToBecome.Data.Tests/Services/UserRepositoryTests.cs

```diff
@@ -0,0 +1,150 @@
+﻿using ApplyToBecome.Data.Services;
+using ApplyToBecome.Data.Services.Interfaces;
+using ApplyToBecome.Data.Tests.AutoFixture;
+using AutoFixture;
+using AutoFixture.Xunit2;
+using Microsoft.Graph;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace ApplyToBecome.Data.Tests.Services
+{
+	public class UserRepositoryTests
+	{
+		private readonly Fixture _fixture = new Fixture();
+
+		[Theory, AutoMoqData]
+		public async Task GetUsers_MatchesGivenName_ReturnsUsers([Frozen] Mock<IGraphUserService> graphUserService,
+			UserRepository sut)
+		{
+			var users = new List<User>();
+
+			for (int i = 0; i < 20; i++)
+			{
+				users.Add(
+					new User { GivenName = _fixture.Create<string>(), Surname = _fixture.Create<string>(), Mail = _fixture.Create<string>(), Id = _fixture.Create<string>() }
+					);
+			}
+
+			users.First().GivenName = "Penelope";
+			users.Last().GivenName = "Peter";
```

Comment:
Probably worth pulling out this login into a GenerateUser(firstGivenName, lastGivenName)

Reply:
Have addressed this.